wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/AddParticipantActionHandler.swift

```diff
@@ -134,31 +136,59 @@ class AddParticipantActionHandler: ActionHandler<AddParticipantAction> {
 
         switch response.httpStatus {
         case 200:
-
             guard
                 let payload = response.payload,
                 let updateEvent = ZMUpdateEvent(fromEventStreamPayload: payload, uuid: nil)
             else {
                 Logging.network.warn("Can't process response, aborting.")
-                action.notifyResult(.failure(.unknown))
+                action.fail(with: .unknown)
                 return
             }
 
             eventProcessor.processConversationEvents([updateEvent])
-
-            action.notifyResult(.success(Void()))
+            action.succeed()
 
         case 204:
-            action.notifyResult(.success(Void()))
-        default:
-            if response.httpStatus == 403 {
-                // Refresh user data since this operation might have failed
-                // due to a team member being removed/deleted from the team.
-                let users: [ZMUser]? = action.userIDs.existingObjects(in: context)
-                users?.filter(\.isTeamMember).forEach({ $0.refreshData() })
+            action.succeed()
+
+        case 403:
+            // Refresh user data since this operation might have failed
+            // due to a team member being removed/deleted from the team.
+            let users: [ZMUser]? = action.userIDs.existingObjects(in: context)
+            users?.filter(\.isTeamMember).forEach({ $0.refreshData() })
+
+            action.fail(with: ConversationAddParticipantsError(response: response) ?? .unknown)
+
+        case 523:
+            guard let failure = Payload.ResponseFailure(response, decoder: decoder),
+                  failure.label == .unreachableDomains,
+                  let unreachableDomains = failure.data?.domains,
+                  let participants = action.userIDs.existingObjects(in: context) as? [ZMUser]
+            else {
+                return action.fail(with: .unknown)
+            }
+
+            let users = participants.belongingTo(domains: unreachableDomains)
+
+            guard !users.isEmpty else {
+                return action.succeed()
             }
+            action.fail(with: .unreachableUsers(users))
```

Comment:
I feel like using guard paired with that inverse condition is a little confusing. The double negation is a bit hard to process. Maybe using an `if else` statement is a bit clearer.

Also renaming `users` to `unreachableUsers` could be a little more explicit

```suggestion
            let unreachableUsers = participants.belongingTo(domains: unreachableDomains)

            if unreachableUsers.isEmpty {
                action.succeed()
            } else {
                action.fail(with: .unreachableUsers(unreachableUsers))
            }   
```